Workflows/Rails upgrades.md

```diff
@@ -0,0 +1,273 @@
+
+##### Gemfile changes
+The Gemfile contains your Rails version requirement (e.g. `gem rails, '~> 6.0.3'`), but the exact version of you're using is noted in `Gemfile.lock`. If the requirement is too restrictive then you will need to relax it (e.g. `gem rails, '~> 6.0'`) before running the bundle update command
+
+```
+bundler update rails --conservative
+```
+
+#####  Configuration file changes
+Updating Rails configuration files updated is sometimes tricky due to the amount of configuration file changes. The interactive update tool `rails app:update` is the best resource to to guide you through the process with well established options like overwrite, show diff and run merge tool. While this process will run smoothly for vanilla apps, you will run into merge conflicts with apps that have a lot of custom edits in the configuration files. Approach each file change separately and make sure you do this step right. You can use [Rails diff](http://railsdiff.org) to get a better (cleaner) view at the before and after state of the configuration files. If you wish to make your next upgrade easier then make sure to add custom configuration options at the end of the environment file (e.g. `config/environments/production.rb`) to reduce the chance of merge conflicts.
+
+The update command may also generate migrations (e.g. `activestorage`) and a new framework defaults file (`config/initializers/new_framework_defaults_x_y.rb`) which you will tackle [later](#3-enabling-rails-x-defaults).
+
+##### Deprecation warnings
+Running the test suite as this point is likely to surface some new deprecation warnings. Addressing these deprecation warnings is crucial since they will start raising runtime errors in the next release cycle.
+
+#### 3. Enabling Rails X defaults
+Most Rails upgrades introduce some amount of breaking changes. The purpose of the `new_framework_defaults_x_y.rb` file is to allow you to **opt-in** to these changes **at your own pace**, separating the upgrade of Rails and introduction of breaking behaviour.
+
+The defaults file should be carefully triaged. You should be able to answer the following set of question before enabling each of the options:
+
+- how does the behavior change look in practice?
+- which parts of the application code will this affect?
+- how would we verify if this change would cause us an issue?
+- what is the estimated effort of changing the application level code to comply with this option?
+
+If the risk or effort of an individual option is deemed too high, then you should tackle it separately from the test of the options. Most of these low hanging fruit defaults can be packaged and deployed to production together, with risky ones being tackled in separate tasks since they usually require some application level changes and a more focused QA effort.
```

Comment:
```suggestion
If the risk or effort of an individual option is deemed too high, then you should tackle it separately from the rest of the options. Most of these low-hanging-fruit defaults can be packaged and deployed to production together, with risky ones being tackled in separate tasks since they usually require some application level changes and a more focused QA effort.
```